core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/public/modeler.xul

```diff
@@ -44,58 +52,65 @@
               <treechildren></treechildren>
             </treeitem>
           </treechildren>
-
         </tree>
-
       </editpanel>
 
       <deck id="modelDeck" flex="1" height="465">
-
         <tabbox id="modelTabbox" flex="1" width="410">
           <tabs>
-            <tab id="analysisTab" label="${analysisTabLabel}" onclick="modeler.setModelerPerspective('ANALYSIS')"></tab>
+            <tab id="analysisTab" label="${analysisTabLabel}"
+                 onclick="modeler.setModelerPerspective('ANALYSIS')"></tab>
             <tab id="reportingTab" label="${reportingTabLabel}"
                  onclick="modeler.setModelerPerspective('REPORTING')"></tab>
           </tabs>
+
           <tabpanels>
             <tabpanel id="analysisTabPanel" height="409">
-
-              <vbox id="analyzerModelPanel" height="411" flex="1" spacing="0" padding="0">
-                <toolbar id="schema-toolbar" mode="icons">
+              <vbox id="analyzerModelPanel" height="411" flex="1"
+                    spacing="0" padding="0" pen:classname="model-deck-panel">
+                <toolbar id="schema-toolbar" mode="icons" flex="0">
                   <toolbarbutton id="expand-all" image="images/spacer.gif" onclick="modeler.expandAll()"
-                                 disabled="false" tooltiptext="${expand_all}" pen:classname="pentaho-expandallbutton"/>
+                                 disabled="false" tooltiptext="${expand_all}"
+                                 pen:classname="pentaho-expandallbutton icon-zoomable"/>
                   <toolbarbutton id="collapse-all" image="images/spacer.gif" onclick="modeler.collapseAll()"
                                  disabled="false" tooltiptext="${collapse_all}"
-                                 pen:classname="pentaho-collapseallbutton"/>
-                  <toolbarspacer flex="1"/>
+                                 pen:classname="pentaho-collapseallbutton icon-zoomable"/>
+                  <toolbarspacer flex="1" pen:classname="flex-auto"/>
```

Comment:
This component call the `setFlex` method also setting `isFlexSpecified` to `true`, but then it doesn't seem to call the `layout` method 🤔 

Then I also tried hardcoding the value to see what happens, but there are some rules that are overriding the one that relies on `--flex-item` 😞 

So for now I feel like it could stay like this and also solve this issue in the second PR, the same one mentioned in a [previous comment](https://github.com/pentaho/data-access/pull/1207#discussion_r1214249729)

Reply:
Some components override the layout method and don't even call the base implementation, as they have a completely different handling of it.